### PR TITLE
Add new synchronizations search end point for pluto

### PR DESCRIPTION
**Pull request by acarrera42**

### CODE REVIEW

**Comment by vmanohar1** [comment]
Will review..

File: app/controllers/carto/api/synchronizations_controller.rb [hunk 1]
```diff
@@ -26,6 +26,22 @@ def index
         head(404)
       end
 
+      def search
+        # Look up all synchronizations that share the same sync filename
+        synchronizations = Carto::Synchronization.where('user_id = ? AND url ~* ?',
+                                                        current_user.id,
+                                                        "\\/#{params[:name]}\\.[\\w\\.]*$")
```
Comment (vmanohar1) [inline app/controllers/carto/api/synchronizations_controller.rb:33]: Can you give one example for 2nd parameter (for URL)?

Comment (vmanohar1) [reply app/controllers/carto/api/synchronizations_controller.rb:33]: I want to revew params[:name] carefully. Rest of this code piece looks ok.

File: config/routes.rb [hunk 2]
```diff
@@ -390,6 +390,7 @@
     delete '(/user/:user_domain)(/u/:user_domain)/api/v1/maps/:map_id/layers/:id'   => 'layers#map_destroy', as: :api_v1_maps_layers_destroy
 
     get    '(/user/:user_domain)(/u/:user_domain)/api/v1/synchronizations'                => 'synchronizations#index',     as: :api_v1_synchronizations_index
+    get    '(/user/:user_domain)(/u/:user_domain)/api/v1/synchronizations/search/:name'   => 'synchronizations#search',    as: :api_v1_synchronizations_search
```
Comment (vmanohar1) [inline config/routes.rb:393]: this looks ok..

File: spec/requests/carto/api/synchronizations_controller_spec.rb [hunk 3]
```diff
@@ -88,7 +338,7 @@
         last_response.status.should == 200
 
         response = JSON.parse(last_response.body)
-        response.fetch('state').should == 'created'
+        response.fetch('state').should == 'queued'
```
Comment (vmanohar1) [inline spec/requests/carto/api/synchronizations_controller_spec.rb:341]: We had one issue case I recall due to created state!

Comment (vmanohar1) [reply spec/requests/carto/api/synchronizations_controller_spec.rb:341]: So this is a good change..

Comment (vmanohar1) [reply spec/requests/carto/api/synchronizations_controller_spec.rb:341]: I believe that it is in the actual code too (this one being in spec)..

File: app/controllers/carto/api/synchronizations_controller.rb (hunk 1 above)
Comment (acarrera42) [reply app/controllers/carto/api/synchronizations_controller.rb:33]: params[:name] should be something like 'Underground_Storage".  

Therefore the url would be:

`http://datavis-maps.prod.bloomberg.com/user/<common_data_user>/api/v1/synchronizations/search/Underground_Storage?api_key=<purposefully_left_out_of_example>`

Comment (vmanohar1) [reply app/controllers/carto/api/synchronizations_controller.rb:33]: Discussed with Andrew.. We need to test this for all cases to ensure that it works with all pluto type urls as well.. If not we can tweak this as needed. Andrew is going to get this table data from prod or qa and then test in dev..

**Comment by vmanohar1** [comment]
Quite a few tests were conducted by importing data from prod.. All tests looked good for the new endpoint..

**Pull request merged**